module_utils/foreman_helper.py

```diff
@@ -34,8 +34,9 @@ def __init__(self, argument_spec, **kwargs):
 
     def parse_params(self):
         module_params = self.filter_module_params()
+        state = module_params.pop('state')
```

Comment:
Is state really omnipresent? Don't we have 'action' modules without it?
If yes, we might want to have a `ForemanAnsibleModule` and a `FormemanEntityAnsibleModule`.

Reply:
```
$ grep -L state *.py
foreman_search_facts.py
foreman_setting_facts.py
__init__.py
katello_content_view_publish.py
katello_content_view_version_promote.py
katello_sync_plan.py
katello_sync.py
katello_upload.py
```

Almost every module ;)

Reply:
Did i forget to mention, that the second one inherits from the first one?

Reply:
@mdellweg that was obvious, tho :package: 

@sean797 your grep didn't catch `foreman_setting`, which does not have state either (which sounds like a bug)

Reply:
implemented